src/dmd/access.d

```diff
@@ -593,7 +592,10 @@ public Dsymbol mostVisibleOverload(Dsymbol s)
                  */
                 auto aliasee = ad.toAlias();
                 if (aliasee.isFuncAliasDeclaration || aliasee.isOverDeclaration)
+                {
+                    error(ad.loc, "`alias X = X` not allowed (with `X = %s`)", ad.toChars());
```

Comment:
Why is it not allowed?  What's the fundamental issue?  Also, isn't the a way to print the actual syntax the user wrote in their source code (`s.toChars()` maybe), rather than giving them a syntax pattern?

Reply:
should be fixed with latest commit

the issue is to fix the infinite loop.

> Also, isn't there a way to print the actual syntax the user wrote

alias X=X is more searchable in bug reports